tutorial/stream-client.md

```diff
@@ -0,0 +1,260 @@
+<html>
+  <head>
+    <title>Anbox Streaming SDK - Tutorial</title>
+    <script type="module" src="static/anbox-stream-sdk.js"></script>
+    <style>
+      html,
+      body,
+      #anbox-stream {
+        height: 100%;
+        margin: 0;
+        width: 100%;
+      }
+    </style>
+  </head>
+
+  <body>
+    <div id="anbox-stream"></div>
+    <script>
+      const sessionId = "{{ session_id }}";
+      class Connector {
+        async connect() {
+          var url = `${window.location.origin}/1.0/sessions/${sessionId}/join`;
+          const resp = await fetch(url, { method: "POST" });
+          if (resp === undefined || resp.status !== 200)
+            throw new Error("failed to join session");
+
+          const response = await resp.json();
+          if (response === undefined || response.status !== "success")
+            throw new Error(response.error);
+
+          return {
+            id: response.metadata.id,
+            websocket: response.metadata.url,
+            stunServers: response.metadata.stun_servers,
+          };
+        }
+
+        disconnect() {}
+      }
+      var stream;
+      window.onload = () => {
+        const connector = new Connector();
+        stream = new AnboxStream({
+          connector: connector,
+          targetElement: "anbox-stream",
+          enableStats: false,
+          stream: {
+            audio: true,
+            video: true,
+          },
+        });
+        stream.connect();
+      };
+    </script>
+  </body>
+</html>
+```
+
+Next, let's make a copy of the Anbox Stream Gateway server to allow the service to validate it:
+
+    # We use cat here to ensure gateway.crt has the users permissions
+    sudo cat /var/snap/anbox-cloud-appliance/common/gateway/server.crt > gateway.crt
+
+As a final step, download a copy of the Anbox streaming SDK and place it inside `stream-client/static`
+
+    curl -o stream-client/static/anbox-stream-sdk.js \
+        https://raw.githubusercontent.com/canonical/anbox-streaming-sdk/refs/heads/main/js/anbox-stream-sdk.js
+
+## Run the stream client
+
+To run the stream client, let's set up a virtual environment for Python to install necessary dependencies:
+
+    cd stream-client
+    python3 -m venv .venv
+    . .venv/bin/activate
+
+Now, we can use `pip` to install all required dependencies
+
+    pip3 install -r requirements.txt
+
+To run the stream client service application, we need to set the necessary environment variables and execute the service via flask.
+
+When installing and initialising the appliance, we would have configured the private IP address of the machine running the appliance. The gateway API is accessible on the local listen address of the appliance on port `9031/tcp`. So that's the IP address and port we use in the GATEWAY_URL.
+
+    export GATEWAY_URL=https://1.2.3.4:9031
+    export GATEWAY_API_TOKEN=xxx
+    export GATEWAY_SERVER_CERT="$PWD"/gateway.crt
+    FLASK_APP=service.py python3 -m flask run -h 1.2.3.4 -p 8080
```

Comment:
```suggestion
    FLASK_APP=service.py python3 -m flask --cert=adhoc run -h 1.2.3.4 -p 8080
```

This will give us TLS with an adhoc generated certificate the user has to accept in the browser. Not meant to be used in production but makes us use TLS always.

Reply:
Just to clarify: `--cert=adhoc` is the part that is not meant for production right? Otherwise, this entire example of setting up a stream client is something that users can use in production?

Reply:
No, using a flask server like this is not meant to be used in production. We can for example to link to https://flask.palletsprojects.com/en/stable/deploying/